src/helpers/buffers/ordering_sender.rs

```diff
@@ -111,6 +114,13 @@ impl State {
         self.closed = true;
         Self::wake(&mut self.stream_ready);
     }
+    fn is_idle(&self) -> bool {
+        self.idle
+    }
+
+    fn reset_idle(&mut self) {
```

Comment:
there is only one client for this API, so `is_idle` can just reset the flag

Reply:
Good point!